docs/src/components/ReferenceTable/PropsReferenceAccordion.tsx

```diff
@@ -0,0 +1,188 @@
+
+  if (name.endsWith('Ref')) {
+    return type;
+  }
+
+  if (['children'].includes(name)) {
+    return type;
+  }
+
+  if (['boolean', 'string', 'number'].includes(type)) {
+    return type;
+  }
+
+  if (!type.includes(' | ') || type.split('|').length < 3) {
+    return type;
+  }
+
+  return 'Union';
+}
+
+export async function PropsReferenceAccordion({
+  data,
+  name: partName,
+  // type = 'props',
+  ...props
+}: Props) {
+  const captionId = `${partName}-caption`;
+
+  return (
+    <Accordion.Root aria-describedby={captionId} {...props}>
+      <span id={captionId} style={visuallyHidden} aria-hidden>
+        Component props table
+      </span>
+      <Accordion.HeaderRow>
+        <Accordion.HeaderLabel>Prop</Accordion.HeaderLabel>
+        <Accordion.HeaderLabel className="max-xs:hidden">Type</Accordion.HeaderLabel>
+      </Accordion.HeaderRow>
+      {Object.keys(data).map(async (name, index) => {
+        const prop = data[name];
+
+        const PropType = await createMdxComponent(`\`${prop.type}\``, {
+          rehypePlugins: rehypeSyntaxHighlighting,
+          useMDXComponents: () => ({ code: TableCode }),
+        });
+
+        const shortPropTypeName = getShortPropType(name, prop.type);
+
+        const ShortPropType = await createMdxComponent(`\`${shortPropTypeName}\``, {
+          rehypePlugins: rehypeSyntaxHighlighting,
+          useMDXComponents: () => ({ code: TableCode }),
+        });
+
+        const PropDefault = await createMdxComponent(`\`${prop.required ? '—' : prop.default}\``, {
+          rehypePlugins: rehypeSyntaxHighlighting,
+          useMDXComponents: () => ({ code: TableCode }),
+        });
+
+        const ShortPropTypeWithDefault = await createMdxComponent(
+          `\`${prop.required ? '—' : `(\`default\`: ${prop.default}\` )`}`,
+          {
+            rehypePlugins: rehypeSyntaxHighlighting,
+            useMDXComponents: () => ({ code: TableCode }),
+          },
+        );
+
+        const PropDescription = prop.description
+          ? await createMdxComponent(prop.description, {
+              rehypePlugins: rehypeSyntaxHighlighting,
+              useMDXComponents: () => inlineMdxComponents,
+            })
+          : null;
+
+        const ExampleSnippet = prop.example
+          ? await createMdxComponent(prop.example, {
+              rehypePlugins: rehypeSyntaxHighlighting,
+              useMDXComponents: () => inlineMdxComponents,
+            })
+          : null;
+
+        return (
+          <Accordion.Item>
+            <Accordion.Trigger
+              index={index}
+              aria-label={`prop: ${name}, type: ${shortPropTypeName} ${prop.default !== undefined ? `(default: ${prop.default})` : ''}`}
+              className="max-xs:gap-4 md:gap-4"
+            >
+              <TableCode className="text-navy">{name}</TableCode>
+              {prop.type && (
+                <span className="flex items-baseline gap-4 text-sm">
+                  <ShortPropType />
```

Comment:
I'm on the fence about hiding the detailed type in the details in some cases. For `render` & `className` it makes sense to always hide it, but for other cases (like below) I think it might degrade the UX a bit to have to click to see the full type.

![image](https://github.com/user-attachments/assets/818ba078-0e73-4482-8b91-bea04ad971e2)

Reply:
Nice for mouse users, but keyboard-centric users (which are probably higher in proportion among devs) don't have that, and requiring a hover is not that far from requiring a click. I still think we should use the space if we have the space.